scripts/pylib/twister/twisterlib/testplan.py

```diff
@@ -756,6 +759,11 @@ def apply_filters(self, **kwargs):
                     if ts.platform_exclude and plat.name in ts.platform_exclude:
                         instance.add_filter("In test case platform exclude", Filters.TESTSUITE)
 
+                    if not device_testing \
```

Comment:
why `device_testing` is checked here?

Reply:
Because if Twister is running on a real board, the result shouldn't be affected by its `simulation` type

Reply:
ok, I see. Platforms like `mps2_an385` have QEMU enabled.

Reply:
> Platforms like `mps2_an385` have QEMU enabled.

Improvements like having the ability to use `qemu` instead of `renode` would be nice in this case, would have to change the schema and make additional changes to the Twister for that